MovieNight/ApiClient.swift

```diff
@@ -11,5 +11,6 @@ import Foundation
 protocol ApiClient {
     func fetchGenres(page: Int, completion: @escaping (APIResult<[Genre]>)-> Void)
     func fetchActors(page: Int, completion: @escaping (APIResult<[Actor]>)-> Void)
-    func fetchMovies(page: Int, completion: @escaping (APIResult<[Movie]>)-> Void)
+    func fetchMovies(endpoint: Endpoint, completion: @escaping (APIResult<[Movie]>)-> Void)
```

Comment:
Make the fecthMovies receive page instead of endpoint. As we discussed, only the api client should know about endpoints.